polyval/src/field/backend/pclmulqdq.rs

```diff
@@ -99,11 +93,20 @@ unsafe fn psrldq8(a: __m128i) -> __m128i {
 
 // TODO(tarcieri): _mm256_clmulepi64_epi128 (vpclmulqdq)
 #[target_feature(enable = "pclmulqdq", enable = "sse2", enable = "sse4.1")]
-unsafe fn pclmulqdq(a: __m128i, b: __m128i, op: clmul::PseudoOp) -> __m128i {
-    match op {
-        clmul::PseudoOp::PCLMULLQLQDQ => _mm_clmulepi64_si128(a, b, 0x00),
-        clmul::PseudoOp::PCLMULHQLQDQ => _mm_clmulepi64_si128(a, b, 0x01),
-        clmul::PseudoOp::PCLMULLQHQDQ => _mm_clmulepi64_si128(a, b, 0x10),
-        clmul::PseudoOp::PCLMULHQHQDQ => _mm_clmulepi64_si128(a, b, 0x11),
+unsafe fn pclmulqdq(a: __m128i, b: __m128i, imm: u8) -> __m128i {
```

Comment:
Do we need this separate function? IIUC you can inline this code directly to `Backend` impl. Also `target_feature` looks redundant, since you already check that this module is built with those features enabled. To be safe you could add a compilation error at the top of this module. Or is it a preparation for future CPUID detection?

Reply:
I can definitely look into doing a lot of additional cleanups after #7 lands, including CPUID detection.